otx/algorithms/detection/tasks/inference.py

```diff
@@ -303,11 +303,13 @@ def _init_test_data_cfg(self, dataset: DatasetEntity):
         return data_cfg
 
     def _update_stage_module(self, stage_module):
-        if self._train_type == TrainType.SEMISUPERVISED:
-            if stage_module == "DetectionTrainer":
-                stage_module = "SemiSLDetectionTrainer"
-            elif stage_module == "DetectionInferrer":
-                stage_module = "SemiSLDetectionInferrer"
+        if stage_module in ["DetectionTrainer", "DetectionInferrer"]:
```

Comment:
let's create prefix dict, as Soobee did. Because, when we will have more training types, "if" will be spreading out
![image](https://user-images.githubusercontent.com/68103869/211561462-7e8289d9-1628-433e-9965-49a470d66ec2.png)
Also, I suggest using assert checking "self.train_type": `assert` self.train_type in prefix_dict, f"{self.train_type} do not have own trainer or inferrer"`

Reply:
Revise it, thanks